### ISSUE TITLE
Scrap command is allowed in RUNNING state

### ISSUE THREAD

**Issue by eflumerf** [DUNE-DAQ/appfwk#157]
I ran daq_application a number of times and built up the following table of commands vs. state:

Command  | NONE | INITIAL | CONFIGURED | RUNNING 
---------|--------|---------|-----------------|---------------
 init | To INITIAL | Not Allowed | Not Allowed | Not Allowed 
 conf | Not Allowed | To CONFIGURED | Not Allowed | Not Allowed 
 start | Not Allowed | Not Allowed | To RUNNING | Not Allowed 
 stop | Not Allowed | Not Allowed | Not Allowed | To CONFIGURED 
 pause | Not Allowed | Not Allowed | Not Allowed | OK (Still RUNNING) 
 resume | Not Allowed | Not Allowed | Not Allowed | OK (Still RUNNING) 
 scrap | Not Allowed | Executed OK (should be Not Allowed) | Executed OK (State still CONFIGURED, should be INITIAL) | Executed OK (State still RUNNING) 

Basically, everything worked as expected, except the scrap command was not blocked from executing from states where it should be blocked. Also, after a scrap in the CONFIGURED state, the state was still CONFIGURED (not INITIAL, as I would expect).

**Comment by eflumerf** [comment]
During my testing, sending a `scrap` command from the RUNNING state did result in an application crash.

**Comment by glehmannmiotto** [comment]
Could you pose the json file used?

**Comment by eflumerf** [comment]
Generated using `python -m minidaqapp.fake_app_confgen -n10 -s 1 -t 5 -d ./frames.bin -o '.' minidaq_1proc.json` with the v2.0.0 tag of `minidaqapp`. Crash occurs by executing `init`, `conf`, `start`, `resume`, `scrap` with a few seconds' pause between each command.

**Comment by eflumerf** [comment]
[minidaq_1proc.json.txt](https://github.com/DUNE-DAQ/appfwk/files/6132803/minidaq_1proc.json.txt)

**Comment by glehmannmiotto** [comment]
The problem is in the generation:

Scrap

 {
    "data": {
        "modules": [
            {
                "match": ""
            }
        ]
    },
    "id": "scrap"
}

The command is defined as a basic command without state constraints! Good catch. I’m providing a fix right away

**Comment by glehmannmiotto** [comment]
should be fixed on minidaqapp glm/fix_scrap. @eflumerf Could you please regenerate and check?

**Comment by eflumerf** [comment]
Yes, it works now, and the last row of that table now works as expected.

**Comment by eflumerf** [comment]
This isn't actually an `appfwk` issue, but a `minidaqapp` one. See [this issue](https://github.com/DUNE-DAQ/minidaqapp/issues/22)
